include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,508 @@
+    //VkBool32    sparseBinding;
+    //VkBool32    sparseResidencyBuffer;
+    //VkBool32    sparseResidencyImage2D;
+    //VkBool32    sparseResidencyImage3D;
+    //VkBool32    sparseResidency2Samples;
+    //VkBool32    sparseResidency4Samples;
+    //VkBool32    sparseResidency8Samples;
+    //VkBool32    sparseResidency16Samples;
+    //VkBool32    sparseResidencyAliased;
+    //VkBool32    variableMultisampleRate;
+    bool inheritedQueries = false;
+
+    /* Vulkan 1.1 Core */
+    //VkBool32           storageBuffer16BitAccess;
+    //VkBool32           uniformAndStorageBuffer16BitAccess;
+    //VkBool32           storagePushConstant16;
+    //VkBool32           storageInputOutput16;
+    //VkBool32           multiview;
+    //VkBool32           multiviewGeometryShader;
+    //VkBool32           multiviewTessellationShader;
+    //VkBool32           variablePointersStorageBuffer;
+    //VkBool32           variablePointers;
+    //VkBool32           protectedMemory;
+    //VkBool32           samplerYcbcrConversion;
+    bool shaderDrawParameters = false;
+
+    /* Vulkan 1.2 Core */
+    //VkBool32           samplerMirrorClampToEdge;
+    bool drawIndirectCount = false;
+    //VkBool32           storageBuffer8BitAccess;
+    //VkBool32           uniformAndStorageBuffer8BitAccess;
+    //VkBool32           storagePushConstant8;
+    //VkBool32           shaderBufferInt64Atomics;
+    //VkBool32           shaderSharedInt64Atomics;
+    //VkBool32           shaderFloat16;
+    //VkBool32           shaderInt8;
+    //VkBool32           descriptorIndexing;
+    //VkBool32           shaderInputAttachmentArrayDynamicIndexing;
+    //VkBool32           shaderUniformTexelBufferArrayDynamicIndexing;
+    //VkBool32           shaderStorageTexelBufferArrayDynamicIndexing;
+    //VkBool32           shaderUniformBufferArrayNonUniformIndexing;
+    //VkBool32           shaderSampledImageArrayNonUniformIndexing;
+    //VkBool32           shaderStorageBufferArrayNonUniformIndexing;
+    //VkBool32           shaderStorageImageArrayNonUniformIndexing;
+    //VkBool32           shaderInputAttachmentArrayNonUniformIndexing;
+    //VkBool32           shaderUniformTexelBufferArrayNonUniformIndexing;
+    //VkBool32           shaderStorageTexelBufferArrayNonUniformIndexing;
+    //VkBool32           descriptorBindingUniformBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingSampledImageUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageImageUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingUniformTexelBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageTexelBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingUpdateUnusedWhilePending;
+    //VkBool32           descriptorBindingPartiallyBound;
+    //VkBool32           descriptorBindingVariableDescriptorCount;
+    //VkBool32           runtimeDescriptorArray;
```

Comment:
no idea what that's about, need an explainer

Reply:
ok then without that feature I'd need to do
```glsl
layout(set = 0, binding = 0) uniform sampler2D texures[OBNOXIOUSLY_LARGE_CONSTANT];
```
and without `descriptorBindingPartiallyBound` I'd need to do
```glsl
layout(set = 0, binding = 0) uniform sampler2D texures[REASONABLY_LARGE_CONSTANT];
```
and then have that constant match the actual DescriptorSetLayout's descriptor's count if `descriptorBindingVariableDescriptorCount` is not present?

Reply:
anyway its obviously false in GL XD